network/gossip/libp2p/test/epochtransition_test.go

```diff
@@ -96,25 +112,31 @@ func (ts *EpochTransitionTestSuite) TearDownTest() {
 func (ts *EpochTransitionTestSuite) TestNewNodeAdded() {
 
 	// create the id, middleware and network for a new node
-	ids, mws, nets := generateIDsMiddlewaresNetworks(ts.T(), 1, ts.logger, 100, nil, []*protocol.ReadOnlyState{ts.state}, false)
+	ids, mws, nets := generateIDsMiddlewaresNetworks(ts.T(), 1, ts.logger, 100, nil, false)
 	newMiddleware := mws[0]
 
 	newIDs := append(ts.ids, ids...)
-	newNetworks := append(ts.nets, nets...)
 
-	// increment epoch
-	ts.currentEpoch = ts.currentEpoch + 1
-
-	// update snapshot mock to return the new ids
-	ts.updateSnapshot(ts.currentEpoch, newIDs)
+	// create a new refresher
+	newIDRefresher := ts.generateNodeIDRefreshers(nets)
+	newIDRefreshers := append(ts.idRefreshers, newIDRefresher...)
 
 	// create the engine for the new node
 	newEngine := generateEngines(ts.T(), nets)
 	newEngines := append(ts.engines, newEngine...)
 
+	// increment epoch
+	ts.currentEpoch = ts.currentEpoch + 1
```

Comment:
Should you do `ts.epochQuery.Transition()` here?

Reply:
so I am still not testing transition from one epoch to the other, I am testing changes of epoch phases. If I call `ts.epochQuery.Transition()` it will make the `epochQuery` point to the next epoch while I am testing against the current epoch.

Reply:
Ah I see, I think the variable name is throwing me a bit. If we aren't doing an epoch transition here, `ts.currentEpoch` refers to the counter of the _next_ epoch once we increment it. Maybe we can do:
```go
// update epoch query to return new IDs for next epoch
ts.addEpoch(ts.currentEpoch+1, newIDs)
```